### PR TITLE
Change Trajectory and Frame API to be consistent with HOOMD conventions

### LINKED ISSUE #76: Change Trajectory and Frame API to be consistent with HOOMD conventions
To make glotzformats more user friendly, we should maintain a similar API to HOOMD's.
Things to change:
- `traj.positions` and `frame.positions` -> `traj.position` and `frame.position`
- `traj.orientations` and `frame.orientations` -> `traj.oritentaion` and `frame.orientation`
- `traj.type_ids` -> `traj.typeid` and `frame.typeid`
- `frame.velocities` -> `frame.velocity`

Task list:
- Add dummy `frame.typeid` property? (will be set properly in #75 - we can resolve the branches when merging or address this issue after #75 closed)
- Rename objects in `Frame` and `Trajectory` class
- Add deprecatoin warnings, but let users use old naming conventions (Do we want to have users add an import or line of code to use deprecated APIs? Or just let them access if but have glotzformats spit out a warning when they use it? I'm more inclined to the second option)
- Write/update unit tests

Extension of #73

### PR DESCRIPTION
**Pull request by gaofy95**
## Description

1. Change traj.positions and frame.positions to traj.position and frame.position
2. Change traj.orientations and frame.orientations to traj.orientation and frame.orientation
3. Change traj.velocities and frame.velocities to traj.velocity and frame.velocity
4. Keep the original positions, orientations, velocities and raise deprecation warning 
TO DO LIST:
1. Change warning levels so that they print on the screen. Maybe use package deprecation will resolve this better.
2. Finish updating documentation
3. Resolve issues with type, types, type_ids
## Motivation and Context

Resolves: #76 

## How Has This Been Tested?

Perform unit tests on the updated property names. Make sure the old names return the same value as the updated one. Check that the warning is actually raised.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds or improves functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation update

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/glotzerlab/garnett/blob/master/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [?] I have updated the documentation (if relevant).
- [?] I have added tests that cover my changes (if relevant).
- [x] All new and existing tests passed.
- [ ] I have updated the [credits](https://github.com/glotzerlab/garnett/blob/master/doc/credits.rst).
- [ ] I have updated the [Changelog](https://github.com/glotzerlab/garnett/blob/master/changelog.rst).

### CODE REVIEW

File: garnett/trajectory.py [hunk 1]
```diff
@@ -216,22 +217,29 @@ def _raise_attributeerror(self, attr):
         else:
             return value
 
+    def _deprecation_warning(self, old_attr, new_attr):
+        warnings.warn(
+            "This property was renamed to %s. %s will be removed in version 0.6.3" %
```
[the remark this answers is not in the document]
Comment (bdice) [reply garnett/trajectory.py]: Also it's preferable to use `.format()` instead of `%` to format strings. The `%` way is more of a Python 2 style.

File: garnett/trajectory.py [hunk 2]
```diff
@@ -494,66 +502,102 @@ def types(self, value):
...
-    def positions(self, value):
+    @property
+    def positions(self):
+        """
+        Nx3 array of coordinated for N particles in 3 dimensions.
+        Deprecated alias for position.
+        """
+        self._deprecation_warning('positions', 'position')
+        try:
+            return self.position(self)
```
Comment (bdice) [inline garnett/trajectory.py]: Why is this line necessary? I think it should just return the property, `return self.position`?

File: garnett/trajectory.py [hunk 3]
```diff
@@ -494,66 +502,102 @@ def types(self, value):
...
+        except TypeError:
+            return self.position
+
+    @position.setter
+    def position(self, value):
```
Comment (bdice) [inline garnett/trajectory.py]: Make sure the setter for `positions` still functions.

File: garnett/trajectory.py [hunk 4]
```diff
@@ -494,66 +502,102 @@ def types(self, value):
...
         # Various sanity checks
         try:
             value = np.asarray(value, dtype=self._dtype)
         except ValueError:
-            raise ValueError("Positions can only be set to numeric arrays.")
+            raise ValueError("position can only be set to numeric arrays.")
```
Comment (bdice) [inline garnett/trajectory.py]: Some of these checks should be lifted to a generic function. It doesn't necessarily have to include the attribute name in the error, so it could just be a function of the data.

File: garnett/trajectory.py [hunk 5]
```diff
@@ -494,66 +502,102 @@ def types(self, value):
...
         if not np.all(np.isfinite(value)):
-            raise ValueError("Positions being set must all be finite numbers.")
+            raise ValueError("position being set must all be finite numbers.")
         elif not len(value.shape) == 2 or value.shape[1] != self.box.dimensions:
             raise ValueError("Input array must be of shape (N,{}) where N is the "
```
Comment (bdice) [inline garnett/trajectory.py]: Is this right? I thought 2D systems still had 3D positions, just with z=0. I am pretty sure that is a convention we use across all group software.

File: garnett/trajectory.py [hunk 6]
```diff
@@ -828,8 +872,8 @@ class Trajectory(BaseTrajectory):
     :param dtype: The default data type for trajectory data.
     """
 
-    TRAJ_ATTRIBUTES = ['N', 'type', 'types', 'type_ids', 'positions',
-                       'orientations', 'velocities', 'mass', 'charge',
+    TRAJ_ATTRIBUTES = ['N', 'type', 'types', 'type_ids', 'position',
```
Comment (bdice) [inline garnett/trajectory.py]: This is the list I was talking about earlier in the GSD reader. I would consider splitting this into two lists, one that has arrays like position/orientation/velocity/.../image and the other can have weird things like N, type, types, type_ids unless you can make it all work together.

File: garnett/trajectory.py [hunk 7]
```diff
@@ -905,6 +949,14 @@ def _max_N(self):
         "Returns the size of the largest frame within this trajectory."
         return max((len(f) for f in self.frames))
 
+    def _deprecation_warning(self, old_attr, new_attr):
+        warnings.warn(
+            "This property was renamed to %s. %s will be removed in version 0.6.3" %
```
Comment (bdice) [inline garnett/trajectory.py]: See above.
```suggestion
            "This property was renamed to %s. %s will be removed in version 0.8.0" %
```

File: garnett/trajectory.py [hunk 8]
```diff
@@ -1280,12 +1361,12 @@ def _generate_type_id_array(types, type_ids):
 
 def copyto_hoomd_blue_snapshot(frame, snapshot):
     "Copy the frame into a HOOMD-blue snapshot."
-    if frame.positions is not None:
-        np.copyto(snapshot.particles.position, frame.positions)
-    if frame.orientations is not None:
-        np.copyto(snapshot.particles.orientation, frame.orientations)
-    if frame.velocities is not None:
-        np.copyto(snapshot.particles.velocity, frame.velocities)
+    if frame.position is not None:
```
Comment (bdice) [inline garnett/trajectory.py]: Similar to the GSD reader/writer above, I would simplify this:

```python
for prop in ['position', ...]:
    data = getattr(frame, prop, None)
    if data is not None:
        np.copyto(getattr(snapshot.particles, prop), data)
```

File: garnett/trajectory.py [hunk 9]
```diff
@@ -1280,12 +1361,12 @@ def _generate_type_id_array(types, type_ids):
 
 def copyto_hoomd_blue_snapshot(frame, snapshot):
```
Comment (bdice) [inline garnett/trajectory.py]: While we're breaking APIs, I'd like to change this name and make it create a Snapshot if one is not provided (`snapshot=None`).
```suggestion
def to_hoomd_snapshot(frame, snapshot=None):
```

Comment (bdice) [reply garnett/trajectory.py]: Similar below. Rename the copy-from operation and make it create a frame if one is not provided. (I think that is possible?)

```python
def from_hoomd_snapshot(snapshot, frame=None):
```

File: tests/test_gsdhoomdfilereader.py [hunk 10]
```diff
@@ -414,9 +414,9 @@ def test_properties(self):
                                     dynamic=['attribute', 'property', 'momentum'])
         gsd_writer.dump_state(self.mc)
         prop_map = dict(
-            position='positions',
-            orientation='orientations',
-            velocity='velocities',
+            position='position',
```
Comment (bdice) [inline tests/test_gsdhoomdfilereader.py]: Similar to above, remove all but `angular_momentum`.

**Review by bdice — COMMENTED** [review]
Great work! This is a big refactor, so I wanted to provide some preliminary comments that might help guide the remaining work. I'm really glad you took on this issue. It should help simplify garnett a lot. If you want to leave the `types` changes to a following pull request, that might be a good decision just to keep this PR size reasonable. Otherwise it might get so long that it's hard to review.

If you want an example of the deprecations in signac that I mentioned, see here: https://github.com/glotzerlab/signac/blob/d6ba3c99984f8473b1123e8dea72a472752d782d/signac/contrib/project.py#L409-L410

File: doc/quickstart.rst [hunk 11]
```diff
@@ -144,7 +144,7 @@ This is an example on how to modify frames in-place:
     import garnett as gt
 
     def center(frame):
-        frame.positions -= np.average(frame.positions, axis=0)
+        frame.position -= np.average(frame.position, axis=0)
```
[the remark this answers is not in the document]
Comment (lyrivera) [reply doc/quickstart.rst]: I think we should just go ahead and replace it now. We can wait until after freud's next release.

[the remark this answers is not in the document]
Comment (bdice) [reply doc/quickstart.rst]: Here's the docs for that feature if needed: https://freud.readthedocs.io/en/latest/modules/box.html#freud.box.Box.center

[the remark this answers is not in the document]
Comment (bdice) [reply doc/quickstart.rst]: Here's the docs for that feature if needed: https://freud.readthedocs.io/en/latest/modules/box.html#freud.box.Box.center

File: garnett/trajectory.py (hunk 5 above)
Comment (lyrivera) [reply garnett/trajectory.py]: Yes I agree, this should be fixed. The question is whether are we throwing `ValueError` when `dimensions==3 and z != 0` or not.

File: garnett/trajectory.py (hunk 3 above)
Comment (lyrivera) [reply garnett/trajectory.py]: Seems like the setters for the old property names were removed? If so they should be added back

**Review by lyrivera — CHANGES_REQUESTED** [review]
Left a few additional comments. Thanks for tackling this.

File: doc/quickstart.rst (hunk 11 above)
[the remark this answers is not in the document]
Comment (klywang) [reply doc/quickstart.rst]: Change it in this PR? Or should it be a seperate one to make this easier to review

File: garnett/getarfilereader.py [hunk 12]
```diff
@@ -49,9 +49,9 @@ def read(self):
         raw_frame = _RawFrameData()
         raw_frame.shapedef = collections.OrderedDict()
         prop_map = {
-            'position': 'positions',
-            'orientation': 'orientations',
-            'velocity': 'velocities',
+            'position': 'position',
```
[the remark this answers is not in the document]
Comment (klywang) [reply garnett/getarfilereader.py]: Since we're refactoring these names to match HOOMD's API, we could also change `angular_momentum_quat` to `angmom` and remove `prop_map` entirely

File: garnett/trajectory.py (hunk 4 above)
Comment (klywang) [reply garnett/trajectory.py]: Is this something to be fixed in this PR or a seperate one?

File: garnett/trajectory.py (hunk 5 above)
Comment (klywang) [reply garnett/trajectory.py]: Is this for this PR as well? Or a seperate issue?

File: doc/quickstart.rst (hunk 11 above)
[the remark this answers is not in the document]
Comment (lyrivera) [reply doc/quickstart.rst]: You can do it here, is a small enough change.

**Comment by lyrivera** [comment]
@klywang @gaofy95 Make sure you also update the [README.md](https://github.com/glotzerlab/garnett/blob/master/README.md) file.

**Comment by lyrivera** [comment]
@klywang @gaofy95 Please let @bdice or me if you need further help with this.

**Comment by lyrivera** [comment]
@klywang @gaofy95 Would it be possible to finish this PR this week?

File: garnett/gsdhoomdfilewriter.py [hunk 13]
```diff
@@ -128,15 +128,15 @@ def write(self, trajectory, stream):
                 except AttributeError:
                     pass
                 try:
-                    snap.particles.position = frame.positions
```
[the remark this answers is not in the document]
Comment (klywang) [reply garnett/gsdhoomdfilewriter.py]: I couldn't find any lists of properties in `trajectory.py` so I made one.

File: doc/quickstart.rst [hunk 14]
```diff
@@ -75,16 +75,16 @@ The actual trajectory data is then either accessed on a *per trajectory* or *per
 Trajectory array access
 -----------------------
 
-Access positions, orientations and types as coherent numpy arrays, by calling the :py:meth:`~.trajectory.Trajectory.load_arrays` method.
-This method will load the complete trajectory into memory and make positions, orientations and types available via properties:
+Access position, orientation and types as coherent numpy arrays, by calling the :py:meth:`~.trajectory.Trajectory.load_arrays` method.
+This method will load the complete trajectory into memory and make position, orientation and types available via properties:
```
[the remark this answers is not in the document]
Comment (lyrivera) [reply doc/quickstart.rst]: @gaofy95  @klywang Seems like you missed this suggestion.

File: garnett/gsdhoomdfilewriter.py [hunk 15]
```diff
@@ -123,46 +123,21 @@ def write(self, trajectory, stream):
...
+                        'position',
+                        'orientation',
+                        'velocity',
+                        'mass',
+                        'charge',
+                        'diameter',
+                        'moment_intertia',
+                        'angmom',
+                        'image'
+                        ]:
```
Comment (lyrivera) [inline garnett/gsdhoomdfilewriter.py]: Since a trajectory object is being passed to this function, this line can be further simplified as
```python
for  prop in trajectory.TRAJ_ATTRIBUTES[4:]:
```

File: doc/quickstart.rst (hunk 14 above)
[the remark this answers is not in the document]
Comment (klywang) [reply doc/quickstart.rst]: @lyrivera Fixed it here b7fae56

File: garnett/gsdhoomdfilewriter.py (hunk 15 above)
Comment (klywang) [reply garnett/gsdhoomdfilewriter.py]: Resolved in 7ac0f68

File: garnett/posfilewriter.py [hunk 16]
```diff
@@ -118,12 +118,12 @@ def _write(msg, end='\n'):
             # Orientations must be provided for all particles
             # If the frame does not have orientations, identity quaternions are used
             try:
-                orientations = frame.orientations
+                orientation = frame.orientation
```
[the remark this answers is not in the document]
Comment (lyrivera) [reply garnett/posfilewriter.py]: @klywang @gaofy95 missed simplification

File: garnett/trajectory.py [hunk 17]
```diff
@@ -216,22 +217,29 @@ def _raise_attributeerror(self, attr):
...
+            (new_attr, old_attr),
+            DeprecationWarning
+        )
+
     def _raw_frame_to_frame(self, raw_frame, dtype=None):
         """Generate a frame object from a raw frame object."""
         N = len(raw_frame.types)
         ret = FrameData()
 
-        positions = np.asarray(raw_frame.positions, dtype=dtype)
```
[the remark this answers is not in the document]
Comment (lyrivera) [reply garnett/trajectory.py]: @klywang @gaofy95 same here

File: garnett/trajectory.py [hunk 18]
```diff
@@ -262,8 +270,8 @@ def _raw_frame_to_frame(self, raw_frame, dtype=None):
             raw_frame.box_dimensions = raw_frame.box.dimensions
             raw_frame.box = np.asarray(raw_frame.box.get_box_matrix(), dtype=dtype)
         box_dimensions = getattr(raw_frame, 'box_dimensions', 3)
-        ret.positions, ret.velocities, ret.orientations, ret.angmom, ret.box = _regularize_box(
-            positions, velocities, orientations, angmom, raw_frame.box, dtype, box_dimensions)
+        ret.position, ret.velocity, ret.orientation, ret.angmom, ret.box = _regularize_box(
+            position, velocity, orientation, angmom, raw_frame.box, dtype, box_dimensions)
         ret.mass = mass
```
[the remark this answers is not in the document]
Comment (lyrivera) [reply garnett/trajectory.py]: @gaofy95 @klywang and here too

File: garnett/trajectory.py (hunk 5 above)
Comment (lyrivera) [reply garnett/trajectory.py]: @bdice what do you think about this?

File: garnett/trajectory.py (hunk 4 above)
Comment (lyrivera) [reply garnett/trajectory.py]: @klywang @gaofy95 You can define a function like the one below 

```python
def _validate_input_array(self, value, size):
    try:
        value = np.asarray(value, dtype=self._dtype)
    except ValueError:
        raise ValueError("This property can only be set to numeric arrays.")
    if not np.all(np.isfinite(value)):
        raise ValueError("Property being set must all be finite numbers.")
    elif not len(value.shape) == 2 or value.shape[1] != size:
        raise ValueError("Input array must be of shape (N,{}) where N is the number of particles.".format(size))
    return value
```
and replace that logic on every setter by calling it:
```python
@orientation.setter
def orientation(self, value):
    self._validate_input_array(value, 4)
    self.load()
    self.frame_data.orientation = value
```

File: garnett/trajectory.py [hunk 19]
```diff
@@ -1301,6 +1410,17 @@ def copyto_hoomd_blue_snapshot(frame, snapshot):
...
+            "This function was renamed to {}. {} will be removed in version 0.8.0.".format(
+                "to_hoomd_snapshot(frame, snapshot)",
+                "copy_to_hoomd_blue_snapshot(frame, snapshot)"
+                ),
+            DeprecationWarning
+            )
+    return to_hoomd_snapshot(frame, snapshot)
+
+
 def copyfrom_hoomd_blue_snapshot(frame, snapshot):
```
Comment (lyrivera) [inline garnett/trajectory.py]: Add deprecation warning here too

File: tests/test_shapes.py [hunk 20]
```diff
@@ -70,6 +70,7 @@ def check_shape_class(self, shape_class):
                 npt.assert_almost_equal(getattr(shape_A, key), value)
                 npt.assert_almost_equal(type_shape[key], value)
 
+
```
Comment (lyrivera) [inline tests/test_shapes.py]: remove this line

File: tests/test_posfilereaderwriter.py [hunk 21]
```diff
@@ -176,9 +176,10 @@ def test_default(self):
             with garnett.read(posfile) as traj:
                 for frame in traj:
                     for name in frame.shapedef.keys():
-                        self.assertEqual(frame.shapedef[name], \
+                        self.assertEqual(frame.shapedef[name],
                                          DEFAULT_SHAPE_DEFINITION)
 
+
```
Comment (lyrivera) [inline tests/test_posfilereaderwriter.py]: remove this line

File: garnett/trajectory.py (hunk 8 above)
Comment (lyrivera) [reply garnett/trajectory.py]: Missed simplification

File: tests/test_posfilereaderwriter.py [hunk 22]
```diff
@@ -164,7 +164,7 @@ def test_injavis_dialect(self):
         self.assert_raise_attribute_error(traj)
 
     def test_default(self):
-        with TemporaryDirectory() as tmp_dir:
+        with TemporaryDirectory() as _:
```
Comment (lyrivera) [inline tests/test_posfilereaderwriter.py]: Revert this.

File: garnett/trajectory.py (hunk 5 above)
Comment (lyrivera) [reply garnett/trajectory.py]: This has been fixed

**Comment by lyrivera** [comment]
@bdice I think this is mostly ready. Can you provide one final review?

File: garnett/trajectory.py [hunk 23]
```diff
@@ -660,14 +634,7 @@ def image(self):
...
     def image(self, value):
-        try:
-            value = np.asarray(value, dtype=np.int32)
-        except ValueError:
-            raise ValueError("Images can only be set to numeric arrays.")
-        if not np.all(np.isfinite(value)):
-            raise ValueError("Images being set must all be finite numbers.")
-        elif not len(value.shape) == 2 or value.shape[1] != 3:
-            raise ValueError("Input array must be of shape (N,3) where N is the number of particles.")
+        value = self._validate_input_array(value, dim=2, nelem=3)
```
Comment (bdice) [inline garnett/trajectory.py]: This would potentially create a bug by attempting to convert the image data to floating point values. The data type of images is int32. I will fix that.

**Review by bdice — APPROVED** [review]
I have done a thorough read-through of this pull request. I found and fixed several bugs, and have suggested some changes to be made in a future pull request. Some changes had unintended consequences:
- the removal of proper type conversions on image attributes, which are of type `int32` but were being converted to floats during "validation"
- the breakage of the plato API (which is not tested on CI because it's difficult to set up enough backends -- we might open an issue to improve this, the plato tests could just run separately using the pre-configured plato CI image)

There were a few docstrings/comments where I saw typos were introduced during this PR in places where the content was previously correct. I think someone's keyboard was a little trigger-happy on the backspace key. 😆 I suggest using `git add -p` or reviewing diffs more carefully before making commits to prevent accidental changes in places like these.

I am approving this PR, but would appreciate a little more caution and "self-review" in future find-replaces / refactorings. That said, I sincerely appreciate your work on this, @klywang and @gaofy95!

File: garnett/trajectory.py [hunk 24]
```diff
@@ -905,6 +872,14 @@ def _max_N(self):
         "Returns the size of the largest frame within this trajectory."
         return max((len(f) for f in self.frames))
 
+    def _deprecation_warning(self, old_attr, new_attr):
+        warnings.warn(
+            "This property was renamed to %s. %s will be removed in version 0.8.0" %
+            (new_attr, old_attr),
+            DeprecationWarning,
+            stacklevel=1
```
[the remark this answers is not in the document]
Comment (bdice) [reply garnett/trajectory.py]: If someone else could test this, I would appreciate it. (It's my bedtime and I wanted to finish this PR review.)

File: garnett/trajectory.py [hunk 25]
```diff
@@ -352,7 +341,7 @@ def make_snapshot(self):
     def copyto_snapshot(self, snapshot):
```
[the remark this answers is not in the document]
Comment (lyrivera) [reply garnett/trajectory.py]: > In this PR, the function `copy_to_hoomd_blue_snapshot` is renamed to `to_hoomd_snapshot`. However, the `Frame` and `FrameData` classes still have methods named `copyto_snapshot`. I propose the following changes:
> 
> * The **module-level function** `to_hoomd_snapshot` should be made into a private module-level function `_to_hoomd_snapshot`. The only reason the module-level function exists here is to provide a method that can be used by both `Frame` and `FrameData` without introducing an inheritance relationship between the two classes.
> * The deprecated **module-level function** `copy_to_hoomd_blue_snapshot` message should be changed to "This function will be **removed** in 0.8.0" and the implementation should be switched from calling `to_hoomd_snapshot` to `_to_hoomd_snapshot`.
> * The **class methods** `copyto_snapshot` of the `Frame` and `FrameData` classes should be renamed to `to_hoomd_snapshot`.
> * The **class methods** `copyto_snapshot` of the `Frame` and `FrameData` classes should raise deprecation warnings and internally call the class method `self.to_hoomd_snapshot`.
> * The **class methods** `make_snapshot` of the `Frame` and `FrameData` classes should be deprecated and removed in 0.8.0. The signature of `to_hoomd_snapshot` should have `snapshot=None` by default, which would return a snapshot object (as was previously done by `make_snapshot`.
> * It does not appear that the **module-level function** `from_hoomd_snapshot` is used or tested anywhere in the package. I would consider making it a private method `_from_hoomd_snapshot`. Similarly, `to_hoomd_snapshot` does not appear to be tested.
> * The deprecated **module-level function** `copyfrom_hoomd_blue_snapshot` message should be changed to "This function will be **removed** in 0.8.0" and the implementation should be switched from calling `from_hoomd_snapshot` to `_from_hoomd_snapshot`.
> 
> I am also unsure whether the `FrameData` class should remain a public class, or whether the `Frame.frame_data` attribute should remain public. I am uncertain what the rationale was for this design. From my experience with the garnett API, it would be sensible to make `FrameData` private as `_FrameData` and to make `Frame.frame_data` into `Frame._frame_data`. In that case, I would also remove the `to_hoomd_snapshot` and other user-facing methods in the `_FrameData` class, which would be purely an internal storage container. Notably, `len(FrameData)` returns the number of types while `len(Frame)` returns the number of particles. Longer-term, it may be a good strategy to unify the `_RawFrameData` and `_FrameData` classes -- I can't see the reason for this class structure.
> 

I have wondered this  myself before and found it quite confusing. I think it is a good idea to review the implementation and unify the classes.

> **I would recommend holding these changes for a follow-up pull request** before the next minor release (since we're aiming to deprecate and break all needed APIs in the same minor release).

File: garnett/trajectory.py (hunk 23 above)
Comment (lyrivera) [reply garnett/trajectory.py]: Nice catch, this was my fault. Thanks for fixing

File: garnett/trajectory.py (hunk 24 above)
[the remark this answers is not in the document]
Comment (lyrivera) [reply garnett/trajectory.py]: @klywang @gaofy95 Can you test this?

File: garnett/trajectory.py (hunk 25 above)
[the remark this answers is not in the document]
Comment (klywang) [reply garnett/trajectory.py]: Just to clarify, does this mean we should revert canges to `copyto_hoomd_blue_snapshot` and `copyfrom_hoomd_blue_snapshot`, then file a seperate issue? Or to leave current changes?

[the remark this answers is not in the document]
Comment (bdice) [reply garnett/trajectory.py]: @klywang You can leave the changes in this PR. Creating an issue describing those changes would be great!

[the remark this answers is not in the document]
Comment (lyrivera) [reply garnett/trajectory.py]: @bdice Just want to make sure I understood correctly: leave the changes to the module level functions we've made thus far here, and create a separate PR for the the rest of the changes?

[the remark this answers is not in the document]
Comment (bdice) [reply garnett/trajectory.py]: @lyrivera Yeah, I think so. We should get this PR finished/merged. Thanks!

[the remark this answers is not in the document]
Comment (lyrivera) [reply garnett/trajectory.py]: @bdice Yes, let's have it merged by the end of the week. @klywang @gaofy95 Whats left?

[the remark this answers is not in the document]
Comment (klywang) [reply garnett/trajectory.py]: Reverted the function names back. There was something about testing the stacklevel of the deprecation warning. We added tests for deprectation but I can't find any documentation about checking stack level in `unittest`. I think we covered everything else.

[the remark this answers is not in the document]
Comment (lyrivera) [reply garnett/trajectory.py]: @klywang  I think the original names of the functions were `copyto_hoomd_blue_snapshot` and `copyfrom_hoomd_blue_snapshot` (without the first underscore)

[the remark this answers is not in the document]
Comment (lyrivera) [reply garnett/trajectory.py]: I reverted the names to the original ones. I'm preparing a separate PR to fix this and address #138.

**Review by lyrivera — APPROVED** [review]
I think this is good to go.
